plugins/ExtendedTools/objprp.c

```diff
@@ -245,6 +258,37 @@ typedef enum _ET_OBJECT_POOLTYPE {
 } ET_OBJECT_POOLTYPE;
 
 #define OBJECT_CHILD_HANDLEPROP_WINDOW 1
+#define OBJECT_CORRECT_HANDLES_COUNT(real_count) ((ULONG)(real_count) - 1)
+
+NTSTATUS EtObjectManagerTrueRef(
+    _In_ HANDLE OwnHandle,
+    _Inout_ PULONG PointerCount
+    )
+{
+    NTSTATUS status = STATUS_INVALID_HANDLE;
+
+#if 0 // enable this then new driver API will available
+    PKPH_PROCESS_HANDLE_INFORMATION handles;
+    if (KsiLevel() >= KphLevelMed &&
+        NT_SUCCESS(status = KsiEnumerateProcessHandles(NtCurrentProcess(), &handles)))
+    {
+        for (ULONG i = 0; i < handles->HandleCount; i++)
+        {
+            if (handles->Handles[i].Handle == OwnHandle)
+            {
+                *PointerCount = *PointerCount - handles->Handles[i].HandleRefCnt - 1;
```

Comment:
Let's please remove this and other areas you have `#if 0` pending driver changes. What you have here isn't correct. An eventual implementation is probably best optimized by letting the driver calculate the real pointer count itself. In order to calculate the real pointer count you need to enumerate all handle tables across the system, not just the current process. The reason you should probably let the driver do this is it can stop enumeration once it has retrieved the `HandleCount` number of handle table entries.

Additionally... I'm not sure if it's even reasonable to be able to calculate the true reference in all cases. The reason being that the real pointer count is a function of the object header pointer count minus the "inverted" ref counts for each active handle. This includes the handles in the CID table (the processes/threads PID/TID table). The CID table is not something immediately available (problem I'm trying to solve in my private branch).

Some background. When the kernel creates a handle it increments the `_OBJECT_HEADER.PointerCount` by `0x8000` and initializes the `_HANDLE_TABLE_ENTRY.RefCnt`. Then when that handle is "used" (e.g. `ObLookupObjectByHandle`) rather than incrementing the `_OBJECT_HEADER.PointerCount` the system simply needs to decrement the `_HANDLE_TABLE_ENTRY.RefCnt`. It then allows the kernel to decrement the `PointerCount` when it has finished with the object. This is "fine" because the `PointerCount` was originally incremented by `0x8000`. When the handle is then closed - the kernel just subtracts the remaining `RefCnt` from the `PointerCount`.

In order to calculate the real pointer count accurately you must enumerate all handle tables on system... the performance of that could be prohibitive of an eventual implementation.

Reply:
Let me think about this a bit more. We can't guarantee we have an exclusive handle to something. And in many cases there will be multiple handles open to the same thing on a system. Then the cases where you could get the true reference wouldn't be many... Sorry to continue to press this, but let's please remove the code that doesn't have a clear accurate path forward for the time being. Showing inaccurate or partial information can be confusing or misleading (current case with the displayed pointer count).

Reply:
7f6029507d943847eecb3b84f7510f43eef0818e
I decide to hide reference count row from object properties window because this value does not currently contain useful and accurate information.
`PhRemoveListViewItem(context->ListViewHandle, PH_PLUGIN_HANDLE_GENERAL_INDEX_REFERENCES);`